fiftyone/utils/eval/coco.py

```diff
@@ -81,188 +87,212 @@ def evaluate_detections(
             :class:`fiftyone.core.labels.Detections` to evaluate
         gt_field ("ground_truth"): the name of the field containing the ground
             truth :class:`fiftyone.core.labels.Detections`
-        save_iou (0.75): an IoU value for which to save per-sample TP/FP/FN
-            counts as top-level sample fields
+        save_ious ([0.5,0.75,0.95]): a list of IoU values for which to compute
+            per-detection and per-image TP/FP/FN
+        sample_iou (0.75): an IoU value for which to save
+            per-sample TP/FP/FN counts as top-level sample fields
     """
     gt_key = "%s_eval" % pred_field
     pred_key = "%s_eval" % gt_field
     eval_id = 0
 
-    try:
-        save_iou_ind = IOU_THRESHOLDS.index(save_iou)
-        save_iou_str = _IOU_THRESHOLD_STRS[save_iou_ind]
-    except ValueError:
-        logger.info(
-            "IoU %f is not in the list of available IoU thresholds: %s",
-            save_iou,
-            IOU_THRESHOLDS,
-        )
-        save_iou_str = None
+    save_ious = list(set(save_ious + [sample_iou]))
+    sample_iou_str = str(sample_iou).replace(".", "_")
+    save_iou_strs = [str(iou).replace(".", "_") for iou in save_ious]
 
     logger.info("Evaluating detections...")
     with fou.ProgressBar() as pb:
         for sample in pb(samples):
-            preds = sample[pred_field]
-            gts = sample[gt_field]
-
-            # Sort preds and gt detections by category label
-            sample_cats = {}
-            for det in preds.detections:
-                det[pred_key] = {}
-                det[pred_key]["ious"] = {}
-                det[pred_key]["matches"] = {
-                    iou_str: {"gt_id": -1, "iou": -1}
-                    for iou_str in _IOU_THRESHOLD_STRS
-                }
-                det[pred_key]["pred_id"] = eval_id
-                eval_id += 1
-                if det.label not in sample_cats:
-                    sample_cats[det.label] = {}
-                    sample_cats[det.label]["preds"] = []
-                    sample_cats[det.label]["gts"] = []
-                sample_cats[det.label]["preds"].append(det)
-
-            for det in gts.detections:
-                det[gt_key] = {}
-                det[gt_key]["matches"] = {
-                    iou_str: {"pred_id": -1, "iou": -1}
-                    for iou_str in _IOU_THRESHOLD_STRS
-                }
+            if samples.media_type == "video":
+                # TODO: is there a better way of doing this?
+                images = [sample.frames[f] for f in sample.frames]
```

Comment:
Use this pattern instead to avoid loading all frames into memory at once:

```py
# Make a generator that will emit `Frame`s one by one
images = sample.frames.values()
```